### PR TITLE
*: AddIndex skip checking unique constraint when TiDB is importing data.

### PR DESCRIPTION
**Pull request by winkyao**
In my mac book, run kvencoder bench, got below results:

```
this pr:
[direct] kv num = 12288000 / kv size = 1362432000
[direct] cost avg = 39.035 sec

master:
[direct] kv num = 12288000 / kv size = 1362432000
[direct] cost avg = 43.753 sec
```

about 10% improvement.

@shenli @coocood PTAL

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package executor_test
 import (
 	"flag"
 	"fmt"
+	"github.com/pingcap/tidb/util/mock"
```
Comment (coocood) [inline executor/executor_test.go]: fix import order

File: table/tables/index_test.go [hunk 2]
```diff
@@ -14,6 +14,7 @@
 package tables_test
 
 import (
+	"github.com/pingcap/tidb/util/mock"
```
Comment (coocood) [inline table/tables/index_test.go]: ditto

File: executor/executor_test.go (hunk 1 above)
Comment (shenli) [inline executor/executor_test.go]: Move this to the next section.

File: table/tables/index_test.go (hunk 2 above)
Comment (shenli) [inline table/tables/index_test.go]: ditto

**Comment by winkyao** [comment]
@shenli @coocood PTAL

**Comment by coocood** [comment]
LGTM

File: table/tables/index.go [hunk 3]
```diff
@@ -177,7 +178,8 @@ func (c *index) GenIndexKey(indexedValues []types.Datum, h int64) (key []byte, d
 // Create creates a new entry in the kvIndex data.
 // If the index is unique and there is an existing entry with the same key,
 // Create will return the existing entry's handle as the first return value, ErrKeyExists as the second return value.
-func (c *index) Create(rm kv.RetrieverMutator, indexedValues []types.Datum, h int64) (int64, error) {
+func (c *index) Create(ctx context.Context, rm kv.RetrieverMutator, indexedValues []types.Datum, h int64) (int64, error) {
+	skipCheck := ctx.GetSessionVars().SkipConstraintCheck
```
Comment (shenli) [inline table/tables/index.go]: This is a dangerous operation. Should we disable the function of [tidb_skip_constraint_check](https://github.com/pingcap/tidb/blob/master/sessionctx/variable/tidb_vars.go#L39) variable.

Comment (winkyao) [reply table/tables/index.go]: maybe we should use individual variable to indicate we are doing importing data?

Comment (shenli) [reply table/tables/index.go]: Yes, such as tidb_importing_data?

Comment (winkyao) [reply table/tables/index.go]: But tidb_skip_constraint_check is skip handle check in https://github.com/pingcap/tidb/blob/f1e89cde1939ea88ec8927f8110da1b5a1b887f5/table/tables/tables.go#L410.

**Comment by shenli** [comment]
LGTM

**Pull request merged**